libgearman-server/plugins/queue/redis/queue.cc

```diff
@@ -124,16 +124,39 @@ bool gearmand::plugins::queue::Hiredis::hmset(vchar_t key, const void *data, siz
 bool gearmand::plugins::queue::Hiredis::fetch(char *key, gearmand::plugins::queue::redis_record_t &req)
 {
   redisContext * context = this->redis();
-  redisReply * reply= (redisReply*)redisCommand(context, "HGETALL %s", key);
+  redisReply * reply = (redisReply*)redisCommand(context, "HGETALL %s", key);
   if (reply == nullptr)
     return false;
 
-  // 2 x (key + value)
-  assert(reply->elements == 4);
+  //FIXME remove workaround
+  if(reply->type == REDIS_REPLY_ERROR) {
+    // workaround to ensure gearmand upgrade.
+    // gearmand <=1.1.15 stores data in string, not in hash.
+    gearmand_log_warning(GEARMAN_DEFAULT_LOG_PARAM, "redis replies for HGETALL: %s", reply->str);
```

Comment:
Is this really a warning? The only time we fetch from the DB is on startup. And this is entirely expected behavior the first time one starts an upgraded gearmand. Seems more like INFO to me.